pyproject.toml

```diff
@@ -28,7 +28,7 @@ Documentation = "https://python-zstandard.readthedocs.io/en/latest/"
 
 [build-system]
 requires = [
-    "cffi>=1.17.0",
+    # "cffi>=1.17.0",  # ok for default, nok for free-threading
```

Comment:
I don't think this is safe to unconditionally disable. Since `cffi` is used in `setup.py`, I believe removing this will have the effect of not producing the cffi backend when you build the package from source unless you disable build isolation and install the cffi dependency yourself.

But, the `requires` entries can't be dynamic and pyproject.toml provides no mechanism for conditional dependencies short of swapping in a custom build backend (I think).

Ugh, Python packaging tooling.

Reply:
I think we can just disable this dynamically in `setup.py` on the free-threaded build - there are ways to check for the free-threaded build at runtime. It's only really a headache if you don't have a `setup.py` and can't do dynamic things. I'll try this along with getting the rust bindings working.

Reply:
> along with getting the rust bindings working

Oh oops, missed that rust [isn't enabled by default yet](along with getting the rust bindings working), so I'll ignore it for now.